### DISCUSSION TITLE
LSP4IJ 0.7.0 Release

### DISCUSSION THREAD

**Discussion by angelozerr** [redhat-developer/lsp4ij#572] [Announcements]
Hi,

we are about to release LSP4IJ 0.7.0. Highlights of this release include 

 * the [LSP client features API](https://github.com/redhat-developer/lsp4ij/blob/main/docs/LSPApi.md)
 * the documentSelector in [registerCapability](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#client_registerCapability)

With @fbricon we decided to do this release before implementing https://github.com/redhat-developer/lsp4ij/discussions/571

More details in the <a href="https://github.com/redhat-developer/lsp4ij/milestone/7?closed=1">changelog</a>.
 
Please give a try to the [latest nightly build](https://plugins.jetbrains.com/plugin/23257-lsp4ij/versions/nightly).

Please note don't override https://github.com/redhat-developer/lsp4ij/blob/2abcd9fffa2e647c4f5fb0face0239f6d4fffb99/src/main/java/com/redhat/devtools/lsp4ij/client/features/LSPCompletionFeature.java#L215 for completion because I need to revisit this method.

Unless you report a blocking issue, we'll proceed with the release next Tuesday 15/10/2024.
 
 cc @CppCXY @InSyncWithFoo @seachicken @KUGDev @AlexWeinstein92 @aparnamichael @MituuZ @turkeylurkey @mrglavas @koxudaxi @FalsePattern @onriv @kristofvb @kdvolder

**Comment by InSyncWithFoo** [comment]
This is not exactly a blocking issue, but the verifier is [yelling at me](https://github.com/InSyncWithFoo/ryecharm/actions/runs/11309833942/job/31454812558) that the new `LSP*Feature` classes cannot be found (I'm targeting PyCharm 2024.3 EAP, if that makes a difference):

```text
Compatibility problems (1): 
    #Package 'com.redhat.devtools.lsp4ij.client.features' is not found
        Package 'com.redhat.devtools.lsp4ij.client.features' is not found along with its 5 classes.
        Probably the package 'com.redhat.devtools.lsp4ij.client.features' belongs to a library or dependency that is not resolved by the checker.
        It is also possible, however, that this package was actually removed from a dependency causing the detected problems. Access to unresolved classes at runtime may lead to **NoSuchClassError**.
        The following classes of 'com.redhat.devtools.lsp4ij.client.features' are not resolved:
          Class com.redhat.devtools.lsp4ij.client.features.LSPCodeActionFeature is referenced in
            insyncwithfoo.ryecharm.ruff.lsp4ij.CodeActionFeature
            insyncwithfoo.ryecharm.ruff.lsp4ij.CodeActionFeature.<init>()
            insyncwithfoo.ryecharm.ruff.lsp4ij.CodeActionFeature.getText(CodeAction codeAction) : String
            insyncwithfoo.ryecharm.ruff.lsp4ij.RuffServerFactory.createClientFeatures() : LSPClientFeatures
          Class com.redhat.devtools.lsp4ij.client.features.LSPDiagnosticFeature is referenced in
            insyncwithfoo.ryecharm.ruff.lsp4ij.DiagnosticFeature
            insyncwithfoo.ryecharm.ruff.lsp4ij.DiagnosticFeature.<init>()
            insyncwithfoo.ryecharm.ruff.lsp4ij.DiagnosticFeature.getHighlightSeverity(Diagnostic diagnostic) : HighlightSeverity
            insyncwithfoo.ryecharm.ruff.lsp4ij.RuffServerFactory.createClientFeatures() : LSPClientFeatures
          Class com.redhat.devtools.lsp4ij.client.features.LSPFormattingFeature is referenced in
            insyncwithfoo.ryecharm.ruff.lsp4ij.FormattingFeature
            insyncwithfoo.ryecharm.ruff.lsp4ij.FormattingFeature.<init>()
            insyncwithfoo.ryecharm.ruff.lsp4ij.RuffServerFactory.createClientFeatures() : LSPClientFeatures
          Class com.redhat.devtools.lsp4ij.client.features.LSPHoverFeature is referenced in
            insyncwithfoo.ryecharm.ruff.lsp4ij.HoverFeature
            insyncwithfoo.ryecharm.ruff.lsp4ij.HoverFeature.<init>()
            insyncwithfoo.ryecharm.ruff.lsp4ij.RuffServerFactory.createClientFeatures() : LSPClientFeatures
          Class com.redhat.devtools.lsp4ij.client.features.LSPClientFeatures is referenced in
            insyncwithfoo.ryecharm.ruff.lsp4ij.RuffServerFactory.createClientFeatures() : LSPClientFeatures
```

**Reply by FalsePattern** [reply]
This is normal, the plugin verifier only verifies against non-nightly marketplace plugin versions, so you'll need to wait until 0.7.0 is fully released for it to pass.

**Comment by angelozerr** [comment]
I launched the release. We need to wait for the JetBrains approval.

**Reply by fbricon** [reply]
0.7.0 has been approved and is now available on the marketplace. Thanks @angelozerr!
